docs/developer_guidelines.md

```diff
@@ -124,68 +125,79 @@ Resources:
 - [One Idea is One Commit] - Interesting committing strategy, comparable to
   atomic commits.
 
-Use the `gitlint` tool to check your commit messages. See
-[tools/gitlint/README.md] for more information.
+### Message style and convention
 
-### Message style
+We follow [Conventional Commits] with a minor deviation:
 
-There are two possible styles for the commit messages: short and full.
+- since we do not usually create libraries, everything related to the
+  `BREAKING-CHANGE` and `!` rule is ignored. Specifically ignore rules 11., 12.,
+  13., 15., and 16. of the Conventional Commits [specification].
 
-#### Short
+[specification]: https://www.conventionalcommits.org/en/v1.0.0/#specification
 
-Short commit messages should be around 50 characters long, the maximum is 72.
+We use [`commited`] tool (run by the `pre-commit`) to validate commit messages.
+For relevant configuration see [`commited.toml`] file in our template
+repository.
 
-Use this when the change that you made is either:
+[`commited`]: https://github.com/crate-ci/committed
+[`commited.toml`]:
+  https://github.com/IRNAS/irnas-zephyr-template/blob/main/commited.toml
 
-- self-explanatory, for example: `Fix typo in README.md` or
-- there is a GitHub issue that provides sufficient context for the change made.
+#### Structure and rules
```

Comment:
consider adding some examples of well written commit messages

Reply:
I have added some commit messages, check if they make sense